zcash_client_sqlite/src/wallet.rs

```diff
@@ -1883,7 +1880,8 @@ pub(crate) fn get_wallet_summary<P: consensus::Parameters>(
     let recover_until_height = recover_until_height(tx)?;
 
     let fully_scanned_height = block_fully_scanned(tx, params)?.map(|m| m.block_height());
-    let summary_height = (chain_tip_height + 1).saturating_sub(std::cmp::max(min_confirmations, 1));
+    let summary_height =
+        (chain_tip_height + 1).saturating_sub(u32::from(confirmations_policy.trusted));
```

Comment:
It's not sufficient merely to change the summary height; the balance is split between confirmed and unconfirmed value, and so the confirmations policy values need to be taken into account when adding up the value of the wallet's notes.